iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/CreateCouplingDevice.java

```diff
@@ -0,0 +1,152 @@
+        BusbarSection bbs1 = network.getBusbarSection(bbsId1);
+        BusbarSection bbs2 = network.getBusbarSection(bbsId2);
+        if (failBbs(bbs1, bbs2, reporter, throwException)) {
+            return;
+        }
+
+        VoltageLevel voltageLevel1 = bbs1.getTerminal().getVoltageLevel();
+        VoltageLevel voltageLevel2 = bbs2.getTerminal().getVoltageLevel();
+        if (voltageLevel1 != voltageLevel2) {
+            LOGGER.error("Busbar sections {} and {} are in two different voltage levels.", bbsId1, bbsId2);
+            busbarsInDifferentVoltageLevels(reporter, bbsId1, bbsId2);
+            if (throwException) {
+                throw new PowsyblException(String.format("Busbar sections %s and %s are in two different voltage levels.", bbsId1, bbsId2));
+            }
+            return;
+        }
+        if (switchPrefixId == null) {
+            switchPrefixId = voltageLevel1.getId();
+        }
+        int breakerNode1 = voltageLevel1.getNodeBreakerView().getMaximumNodeIndex() + 1;
+        int breakerNode2 = breakerNode1 + 1;
+        int bbs1Node = bbs1.getTerminal().getNodeBreakerView().getNode();
+        int bbs2Node = bbs2.getTerminal().getNodeBreakerView().getNode();
+
+        createNBBreaker(breakerNode1, breakerNode2, "", switchPrefixId, voltageLevel1.getNodeBreakerView(), false);
+        createNBDisconnector(bbs1Node, breakerNode1, "_" + bbs1Node, switchPrefixId, voltageLevel1.getNodeBreakerView(), false);
+        createNBDisconnector(bbs2Node, breakerNode2, "_" + bbs2Node, switchPrefixId, voltageLevel1.getNodeBreakerView(), false);
+
+        BusbarSectionPosition position1 = bbs1.getExtension(BusbarSectionPosition.class);
+        BusbarSectionPosition position2 = bbs2.getExtension(BusbarSectionPosition.class);
+        int nbOpenDisconnectors = 0;
+        if (position1 != null) {
+            if (position2 != null) {
+                List<BusbarSection> bbsList1 = voltageLevel1.getNodeBreakerView().getBusbarSectionStream()
+                        .filter(b -> b.getExtension(BusbarSectionPosition.class) != null)
+                        .filter(b -> b.getExtension(BusbarSectionPosition.class).getSectionIndex() == position1.getSectionIndex())
+                        .filter(b -> !b.getId().equals(bbsId1)).collect(Collectors.toList());
+                List<BusbarSection> bbsList2 = voltageLevel2.getNodeBreakerView().getBusbarSectionStream()
+                        .filter(b -> b.getExtension(BusbarSectionPosition.class) != null)
+                        .filter(b -> b.getExtension(BusbarSectionPosition.class).getSectionIndex() == position2.getSectionIndex())
+                        .filter(b -> !b.getId().equals(bbsId2)).collect(Collectors.toList());
+                if (bbsList1.size() != 1 || position1.getSectionIndex() != position2.getSectionIndex()) { // if both busbar sections not in same section or in same section with other busbar sections
```

Comment:
The comment here is not really useful, it just reads what is in the if. I think it would be clearer to explain why we don't connect the other busbar in the particular case.

Reply:
Yes but the issue is if you call any hypotheses more than once (this one in particular but this is valid for all sort of hypotheses), you run the risk of ID collision...